kivy/core/image/_img_sdl2.pyx

```diff
@@ -42,53 +42,58 @@ def save(filename, w, h, fmt, pixels, flipped):
 cdef load_from_surface(SDL_Surface *image):
     cdef SDL_Surface *image2 = NULL
     cdef SDL_Surface *fimage = NULL
-    cdef SDL_PixelFormat pf
+    cdef int want_rgba = 0, target_fmt = 0, n = 0
     cdef bytes pixels
 
+    if image == NULL:
+        Logger.warn('ImageSDL2: load_from_surface() with NULL surface')
+        return None
+
+    # SDL 2.0.5 now has endian-agnostic 32-bit pixel formats like RGB24,
+    # but we can't count on that yet (RGB24 is available since 2.0)
+    if SDL_BYTEORDER == SDL_BIG_ENDIAN:
+        want_rgba = SDL_PIXELFORMAT_RGBA8888
+    else:
+        want_rgba = SDL_PIXELFORMAT_ABGR8888
+
+    # Determine if we need pixel format conversion
+    if image.format.format == want_rgba:
+        fmt = 'rgba'
+    elif image.format.format == SDL_PIXELFORMAT_RGB24:
+        fmt = 'rgb'
+    elif image.format.Amask:
+        fmt = 'rgba'
+        target_fmt = want_rgba
+    elif image.format.palette != NULL:
+        # We want RGB24 here, and rgba only if the palette has alpha
+        fmt = 'rgba'
+        target_fmt = want_rgba
+# FIXME: ... but this breaks compile, "SDL_Color has no member unused"
```

Comment:
It's a problem with `unused` naming because the Cython parts used only `rgb` for some reason, but leaving alpha completely probably broke the compilation, therefore `cdef Uint8 unused`. See [this line](https://github.com/kivy/kivy/pull/5252/files#diff-d18c5eedfc97dd5c3934914bc1e8bc8aR92) of my shaped window PR to see the fix. It should work just fine after it.